src/main/fc/cli.c

```diff
@@ -2548,12 +2539,12 @@ static void cliSet(char *cmdline)
             if (strncasecmp(cmdline, valueTable[i].name, strlen(valueTable[i].name)) == 0 && variableNameLength == strlen(valueTable[i].name)) {
 
                 bool changeValue = false;
-                cliVar_t value  = { .uint16 = 0 };
+                cliVar_t value  = { .int16 = 0 };
                 switch (valueTable[i].type & VALUE_MODE_MASK) {
                     case MODE_DIRECT: {
-                            value.uint16 = atoi(eqptr);
+                            value.int16 = atoi(eqptr);
 
-                            if (value.uint16 >= valueTable[i].config.minmax.min && value.uint16 <= valueTable[i].config.minmax.max) {
+                            if (value.int16 >= valueTable[i].config.minmax.min && value.int16 <= valueTable[i].config.minmax.max) {
```

Comment:
Correct. But since `valueTable[i].config.minmax.max` is an int16, even unsigned int16 parameters cannot be set to values outside of the int16 range. Since no parameter currently has a range larger than 32000, I decided that having this restriction is preferable to adding the complexity of making `valueTable[i].config.minmax` a union, for no gain at all.